modules/ROOT/pages/header-inject-remove-task.adoc

```diff
@@ -1,54 +1,76 @@
+>s|Summary      | Adds headers to a request or a response
 >s|Category | Transformation
 >s| First Mule version available | 4.1.0
 .1+>.^s| Returned Status Codes
-|
+
+[%autowidth.spread,cols="a,a"]
+!===
+.1+>.^! 500 ! Configuration includes an expression that was not evaluated.
+! The provided expression in the configuration was not validated. +
+!===
+
 |===
-    
-The Header Injection policy adds headers to the request/response of the message. The Header Removal policy removes all listed headers.
+
+The Header Injection policy adds HTTP headers to the request or response of a message. When you configure this policy for your API, you must specify an inbound and outbound map of the headers that you want to add in the message processing in the form of a key-value pair:
 
 image::header-inject-remove-diagram.png[]
 
+You can optionally include DataWeave expressions for Mule 4.0 and Mule runtime engine (Mule) expressions for Mule 3.0 and in the value of the header:
+
+`“#[attributes.requestPath]”`
 
-== Applying Header Injection Policy
+In this example, all the headers matching the configured expression are injected to the message.
 
-To apply this policy to an API, specify an inbound/outbound map of the headers that you want to add in the message processing in the form of a key-value pair.
 
-image::inbound-header-map.png[]
+== How This Policy Works
 
-image::outbound-header-map.png[]
+In some cases, you might need to xref::header-removal-policy.adoc[remove] a header from a message before you inject the header.
 
+If the injected header already exists in the message attributes, the policy creates a new header without overriding the value of the existing one. This action creates two headers with the same name. To overwrite the value of a header, you must remove it before injecting the header.  
 
-If the injected header already exists in the message attributes, the policy creates a new header, without overriding the value of the existing one. This action creates two headers with the same name.
+[NOTE]
+====
+Comma separated values are not considered multiple values by the Mule HTTP Connector.
+====
 
-*Note:* Comma separated values are not considered multiple values by the Mule HTTP Connector.
 
-You can use DataWeave expressions to set a value. For example:
+// image::inbound-header-map.png[]
 
-`“#[attributes.requestPath]”`
+// image::outbound-header-map.png[]
+
+//Anything to add here?
+
+== Configuring Policy Parameters
+
+When you apply the Header Injection policy to your API from the UI, you can configure the following parameters:
+
+[%header%autowidth.spread,cols="a,a"]
+|===
+| Parameter | Description 
+|Inbound Header Map  | The key-value pair for the inbound header, for example, `in-header(key):3(value)`
+|Outbound Header Map | The key-value pair for the outbound header, for example, `out-header(key):4(value)`
+|Method & Resource conditions |The option to add configurations to only a select few or all methods and resources of the API
```

Comment:
```suggestion
|Method and resource conditions |The option to add configurations to only a select few or all methods and resources of the API
```
...but what does this mean? This is not described earlier, unless it refers to the "optional expressions" that you add using DataWeave (Mule 4.x) or Mule (Mule 3.x). Clarify. 
GLOBALLY avoid referring to something in more than one way; it is very confusing to your audience.

Reply:
Discussed over the call. Leaving it as is.